### PR TITLE
WIP:Feat/es6, es7 support and elastica migration

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |Y|
|New feature?   |Y|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

The idea of this PR is bright es6 and es7 support with a es5 client. In order to have a bridge version to migrate all projets to a ES7 without bing bang

### CODE REVIEW

File: src/Controller/ElasticsearchController.php [hunk 1]
```diff
@@ -7,45 +7,35 @@
 use Elasticsearch\Common\Exceptions\ElasticsearchException;
 use Elasticsearch\Common\Exceptions\Missing404Exception;
 use Elasticsearch\Common\Exceptions\NoNodesAvailableException;
+use EMS\CommonBundle\Elasticsearch\Response\Response as CommonResponse;
```
Comment (Davidmattei) [inline src/Controller/ElasticsearchController.php:10]: Don't like its called CommonResponse, Its a ElasticsearchResponse

File: src/Controller/ElasticsearchController.php [hunk 2]
```diff
@@ -790,121 +749,53 @@ public function searchAction(Request $request)
...
-			      }
-			   }
-			}', true));
-
-            $aggregateOptions = $this->getAggregateOptionService()->getAll();
-            /** @var AggregateOption $option */
-            foreach ($aggregateOptions as $id => $option) {
-                $body['aggs']['agg_' . $id] = $option->getConfigDecoded();
-            }
+            $commonSearch = $searchService->generateSearch($search);
```
Comment (Davidmattei) [inline src/Controller/ElasticsearchController.php:752]:
```suggestion
            $esSearch = $searchService->generateSearch($search);
```

File: src/Controller/ElasticsearchController.php [hunk 3]
```diff
@@ -915,33 +806,64 @@ public function searchAction(Request $request)
...
+        }
         $out = [];
-        foreach ($results['aggregations']['types']['buckets'] as $type) {
-            $out[] = $type['key'];
+        foreach ($aggregation->getBuckets() as $bucket) {
+            $key = $bucket->getKey();
+            if ($key === null) {
+                continue;
+            }
+            $out[] = $key;
```
Comment (Davidmattei) [inline src/Controller/ElasticsearchController.php:866]: May we should add a function $aggregation->getKeys();

File: src/Twig/AppExtension.php [hunk 4]
```diff
@@ -38,6 +41,7 @@
 use Twig\TwigFilter;
 use Twig\Environment as TwigEnvironment;
 use Twig\TwigFunction;
+use \EMS\CommonBundle\Search\Search as CommonSearch;
```
Comment (Davidmattei) [inline src/Twig/AppExtension.php:44]:
```suggestion
use \EMS\CommonBundle\Search\Search;
```

File: src/Service/SearchService.php [hunk 5]
```diff
@@ -2,26 +2,51 @@
 
 namespace EMS\CoreBundle\Service;
 
+use Elastica\Query\BoolQuery;
+use EMS\CommonBundle\Service\ElasticaService;
+use EMS\CoreBundle\Entity\Environment;
 use EMS\CoreBundle\Entity\Form\Search;
-use EMS\CoreBundle\Entity\Form\SearchFilter;
+use EMS\CommonBundle\Search\Search as CommonSearch;
```
Comment (Davidmattei) [inline src/Service/SearchService.php:9]:
```suggestion
use EMS\CommonBundle\Search\Search;
```
Call the other search import EntitySearch

**Review by Davidmattei — CHANGES_REQUESTED** [review]
Nice! Just had some remarks about naming things 👍

File: src/Controller/ElasticsearchController.php (hunk 1 above)
Comment (theus77) [reply src/Controller/ElasticsearchController.php:10]: Me neither but there is another use Symfony\Component\HttpFoundation\Response;

File: src/Controller/ElasticsearchController.php (hunk 2 above)
Comment (theus77) [reply src/Controller/ElasticsearchController.php:752]: Renamed as suggested

File: src/Controller/ElasticsearchController.php (hunk 3 above)
Comment (theus77) [reply src/Controller/ElasticsearchController.php:866]: Done

File: src/Twig/AppExtension.php (hunk 4 above)
Comment (theus77) [reply src/Twig/AppExtension.php:44]: We also use EMS\CoreBundle\Entity\Form\Search;

File: src/Service/SearchService.php (hunk 5 above)
Comment (theus77) [reply src/Service/SearchService.php:9]: se also use EMS\CoreBundle\Entity\Form\Search;

File: composer.json [hunk 6]
```diff
@@ -14,6 +14,12 @@
 	"keywords" : [
 		"elasticms"
 	],
+	"repositories": [
+		{
+			"type": "vcs",
+			"url": "https://github.com/ems-project/EMSCommonBundle"
+		}
+	],
```
Comment (Davidmattei) [inline composer.json:22]: can be removed

**Review by Davidmattei — APPROVED** [review]
Approved but first revert composer.json

**Review by Davidmattei — APPROVED** [review]

**Pull request merged**